File: 3B/python_web_service/web_service_vitamins_status.py
#!/usr/bin/python
from flask import Flask, request
from ast import literal_eval
from json import dumps
from flask_cors import CORS
from lib import makeAllWhite, makeAllWhiteStatus
from sys import argv

def getVitamins(request):
    if request.data:
        return literal_eval(request.data)["vitamins"]
    else:
        return request.values.get("vitamins")

app = Flask(__name__)
CORS(app)

# default: return a standard message to show that the service is active


@app.route('/', methods=['POST', 'GET'])
def default():
    return "Hi to everyone. I' am alive!! :)"


# response_moves: create a web service on  "EXECUTION_DOMAIN:PORT/vitamin/moves"
# getting as input the configuration of the vitamins eg: "3B 4B 5G 6W"
# and returning the list of moves to make the colors of the shapes white


@app.route('/vitamin/moves', methods=['POST', 'GET'])
def response_moves():
    s_vitamins = getVitamins(request)
    return dumps(makeAllWhite(s_vitamins), indent=4)

# response_status: create a web service on  "EXECUTION_DOMAIN:PORT/vitamin/status"
# getting as input the configuration of the vitamins eg: "3B 4B 5G 6W" and a list
# and returning the list of status changed during the transition
# to make the colors of the shapes white


@app.route('/vitamin/statuses', methods=['POST', 'GET'])
def response_status():
    s_vitamins = getVitamins(request)
    return dumps(makeAllWhiteStatus(s_vitamins, makeAllWhite(s_vitamins)), indent=4)

if __name__ == '__main__':
    input_port = None
    if len(argv) > 1:
        input_port = argv[1]

    app.run(port=input_port, debug=False, threaded=True)

File: 3B/es6_nodejs_web_service/web_service_vitamins_status.js
const express = require('express');
const cors = require('cors');
const functions = require('./lib/functions');
let app = express();
app.use(cors());

//index: return a standard message to show that the service is active
function index(req, res) {res.send("Hi to everyone. I' am alive!! :)")}


/* response_moves: create a web service on  "EXECUTION_DOMAIN:PORT/vitamin/moves"
 getting as input the configuration of the vitamins eg: "3B 4B 5G 6W"
 and returning the list of moves to make the colors of the shapes white */

function response_moves(req, res) {
    res.send(JSON.stringify(functions.makeAllWhite(req.param("vitamins")),null,null))
}

/* response_status: create a web service on  "EXECUTION_DOMAIN:PORT/vitamin/status"
 getting as input the configuration of the vitamins eg: "3B 4B 5G 6W" and a list
 and returning the list of status changed during the transition
 to make the colors of the shapes white */

function response_status(req, res) {
    let vitamin_s = req.param("vitamins").trim();
    res.send(JSON.stringify(functions.makeAllWhiteStatus(vitamin_s,functions.makeAllWhite(vitamin_s)),null,null))
}
app.post("/",index);
app.get('/', index);
app.post('/vitamin/moves', response_moves);
app.get('/vitamin/moves', response_moves);
app.post('/vitamin/statuses', response_status);
app.get('/vitamin/statuses', response_status);

app.listen(5000);